scripts/build/cc/gcc.sh

```diff
@@ -583,6 +583,9 @@ do_gcc_core_backend() {
         cflags_for_target="${cflags_for_target} -idirafter ${CT_HEADERS_DIR}"
     fi
 
+    # Assume '-O2' by default for building target libraries.
+    cflags_for_target="-g -O2 ${cflags_for_target}"
```

Comment:
I think this should be the default for TARGET_CFLAGS in Kconfig, this will be more obvious to the user. Needs to be tested more thoroughly, though, as it will affect not just gcc build.

Reply:
An argument against that would be that someone like me could expect the flags work identical to "manual" builds.
With empty flags: pick the packages defaults (which might well be different).

Prepending the package defaults seems so be the most sane to me, as you could eg. easily disable debugging info with `-g0` while leaving the respective optimization defaults untouched, or adding `-flto` for adding LTO info.

Then maybe we can fixup all the small deficiencies and differences in packages, see #1267, #1280